src/components/editor/MonacoEditor.tsx

```diff
@@ -138,7 +141,7 @@ function MonacoEditor({
         },
     };
 
-    if (currentCatalog?.spec) {
+    if (currentCatalog.catalog_name && specAsString) {
```

Comment:
We only want to render if we have both the name and the string version of the spec